src/odemis/driver/technolution.py

```diff
@@ -443,18 +533,221 @@ def _mergeMetadata(self):
+            logging.warning("An incorrect shape of the cell translation parameters is provided.\n "
+                            "Please change the shape of the cell translation parameters according to the shape of the "
+                            "MPPC detector.\n "
+                            "Cell translation parameters remain unchanged.")
+            return self.cellTranslation.value
+
+        for row, cellTranslationRow in enumerate(cellTranslation):
+            if len(cellTranslationRow) != self._shape[1]:
+                logging.warning("An incorrect shape of the cell translation parameters is provided.\n"
+                                "Please change the shape of the cellTranslation parameters according to the shape of "
+                                "the MPPC detector.\n "
+                                "Cell translation parameters remain unchanged.")
+                return self.cellTranslation.value
+
+            for column, eff_origin in enumerate(cellTranslationRow):
+                if len(eff_origin) != 2:
+                    logging.warning("Incorrect cell translation parameters provided, wrong number of coordinates for "
+                                    "cell (%s, %s) are provided.\n"
+                                    "Please provide an 'x effective origin' and an 'y effective origin' for this cell "
+                                    "image.\n "
+                                    "Cell translation parameters remain unchanged." %
+                                    (row, column))
+                    return self.cellTranslation.value
+
+                if not isinstance(eff_origin[0], int) or not isinstance(eff_origin[1], int):
+                    logging.warning("An incorrect type is used for the cell translation coordinates of cell (%s, %s).\n"
+                                    "Please use type integer for both 'x effective origin' and and 'y effective "
+                                    "origin' for this cell image.\n"
+                                    "Type expected is: '(%s, %s)' type received '(%s, %s)'\n"
+                                    "Cell translation parameters remain unchanged." %
+                                    (row, column, int, int, type(eff_origin[0]), type(eff_origin[1])))
+                    return self.cellTranslation.value
+                elif eff_origin[0] < 0 or eff_origin[1] < 0:
+                    logging.warning("Please use a minimum of 0 cell translation coordinates of cell (%s, %s).\n"
+                                    "Cell translation parameters remain unchanged." %
+                                    (row, column))
+                    return self.cellTranslation.value
+
+        return cellTranslation
+
+
+    def _setcellDigitalGain(self, cellDigitalGain):
+        if len(cellDigitalGain) != self._shape[0]:
+            logging.warning("An incorrect shape of the digital gain parameters is provided. Please change the "
```

Comment:
Here the same applies as for setCellTranslation. Raise error if value is not correct?